app/src/main/java/amazingme/database/FirebaseHelper.java

```diff
@@ -74,4 +62,62 @@ public static void signOut() {
         getFirebaseAuthInstance().signOut();
     }
 
+    private static void onCompleteRegistrationHandler(final Activity activity, final Task<AuthResult> task) {
+        final ActivityManager activityManager = ActivityManager.getInstance();
+        final Resources res = activity.getResources();
+
+        final String registrationFailed = res.getString(R.string.dialog_registration_failed);
+        final String registrationTag = res.getString(R.string.tag_registration);
+        final AlertDialog.Builder alertDialog = DialogHelper.getAlertDialog(activity, registrationFailed);
+
+        if (task.isSuccessful()) {
+            activityManager.goTo(activity, EnumeratedActivity.MAIN_MENU);
+        } else {
+            String exceptionMessage;
+            try {
```

Comment:
hmmmm this is good functionally... 

what are your thoughts on making a class for this just so it's cleaner and easier to read? 

so maybe like FirebaseAuthExceptionHandler? 

it'll take a few lines of code out that I don't personally* care for in this class, but it's more talking. 

idk. just trying to see how sleek we can make this.

Reply:
If this block of code is not going to be reused I don't really see any point in making a new class.

I do agree that the placement of this code is not ideal, since FirebaseHelper should be just that, a helper for Firebase related functions (instead of showing dialog boxes/handling exceptions). But I can't figure out how to put it in LoginActivity due to the async callback issue.

Let me know what you think. For now we can either:
1. Keep it, or
2. Refactor it to a new class, as you suggested, or
3. Try to figure out the best location to place it (which I currently have no idea).

Reply:
can you do me a favor into just a helper method? 

a class is overkill but i'd like to just let future reviewers see the business side, so to speak. 

"so if the task is successful, go to main menu, otherwise show the correct dialogue", where that last part is inferred through a well-named helper function. 

how's that?